### PR TITLE
Str 4443 kubeshark 38

### PR DESCRIPTION
**Pull request by ericatcorsha**
Before you submit your PR, make sure you check each of the following:

1. [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md) and signed the [Contributor License Agreement](./CLA.md).
2. [x] My code is clean and ready-to-PR, including being free of lint errors and dead code
3. [x] I have **tested** my change / code and am confident that it works
4. [x] I have filled in the template below completely and accurately

##### Changelog

- Upgrade collector to work with Kubeshark 38

##### Testing
```sh
cd ~
rm -rf /tmp/str-4443
mkdir /tmp/str-4443
cd /tmp/str-4443

ln -s ~/src/cast /tmp/str-4443/cast
```

Download version 37 of Kubeshark by following the instructions at <https://github.com/kubeshark/kubeshark/releases/tag/37.0>

```sh
# Intel Mac
curl -Lo kubeshark https://github.com/kubeshark/kubeshark/releases/download/37.0/kubeshark_darwin_amd64 && chmod 755 kubeshark

# M1
curl -Lo kubeshark https://github.com/kubeshark/kubeshark/releases/download/37.0/kubeshark_darwin_arm64 && chmod 755 kubeshark

./kubeshark version
```

Run the old version of CAST

```sh
cd /tmp/str-4443/cast
git checkout 80d0ec9e877ccf33d1770afbc079a64a07337550
export PATH="/tmp/str-4443:$PATH"
# verify that the version of kubeshark is 37.0
kubeshark version
```

Clean up everything

```sh
kubeshark --set kube-context=docker-desktop clean
skaffold delete --kube-context docker-desktop # ignore any errors about missing releases
kubectl --context docker-desktop delete ns cast
kubectl --context docker-desktop delete ns kubeshark
```

Start the test Skaffold

```sh
skaffold run --platform=linux/amd64 --port-forward --kube-context docker-desktop
```

Wait for the following log messages

    Port forwarding service/cast-postgresql in namespace cast, remote port 5432 -> http://127.0.0.1:5432
    Port forwarding service/cast-service in namespace cast, remote port 80 -> http://127.0.0.1:8000
    Port forwarding service/httpbin in namespace cast, remote port 80 -> http://127.0.0.1:8080
    Port forwarding service/cast-postgresql-hl in namespace cast, remote port 5432 -> http://127.0.0.1:5433

Open dashboard

<http://localhost:8000/>

Take a screenshot of the browser to compare with the new version

Kill the Skaffold by pressing Ctrl+C

Test with the latest Kubeshark

Install the latest kubeshark system-wide

<https://docs.kubeshark.co/en/install>

```sh
rm /tmp/str-4443/kubeshark
sh <(curl -Ls https://kubeshark.co/install)
kubeshark version
```

Switch the CAST main branch

```sh
cd /tmp/str-4443/cast
git checkout main
git checkout str-4443-kubeshark-38
```

Verify the we have the correct version of Kubeshark

```sh
kubeshark version # >= 38.5
```

Clean up everything

```sh
kubeshark --set kube.context=docker-desktop clean
skaffold delete --kube-context docker-desktop # ignore any errors about missing releases
kubectl --context docker-desktop delete ns cast
kubectl --context docker-desktop delete ns kubeshark
```

Start the test Skaffold

```sh
skaffold run --platform=linux/amd64 --port-forward --kube-context docker-desktop
```

Wait for the following log messages

    Port forwarding service/cast-postgresql in namespace cast, remote port 5432 -> http://127.0.0.1:5432
    Port forwarding service/cast-service in namespace cast, remote port 80 -> http://127.0.0.1:8000
    Port forwarding service/httpbin in namespace cast, remote port 80 -> http://127.0.0.1:8080
    Port forwarding service/cast-postgresql-hl in namespace cast, remote port 5432 -> http://127.0.0.1:5433

Open dashboard

<http://localhost:8000/>

Compare what you see to the screenshot you made with the old version

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -295,35 +348,39 @@ func requiredEnv(envName string) string {
 	return ret
 }
 
-// handleMessage takes a message read from the kubeshark websocket and processes it for insertion
+// handleTrafficItem takes a message read from the kubeshark websocket and processes it for insertion
 // into the postgres database. It returns the processed message and a CASTMetadata struct for the message.
-func handleMessage(message []byte, msgStruct *Message) ([]byte, []byte, error) {
+func handleTrafficItem(trafficItem []byte, itemStruct *TrafficItem) ([]byte, []byte, error) {
```
Comment (DavidCorsha) [inline collector/collector.go:353]: This naming is a little hard to follow, I would suggest calling the struct the trafficItem
```suggestion
func handleTrafficItem(messageJSON []byte, trafficItem *TrafficItem) ([]byte, []byte, error) {
```

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: collector/collector.go [hunk 2]
```diff
@@ -242,26 +258,48 @@ func exportRecords(pgConnection *sql.DB, ksConnection *websocket.Conn, ctx conte
 	}
 }
 
-func writeRecords(pgConnection *sql.DB, ksConnection *websocket.Conn) error {
-	_, message, err := ksConnection.ReadMessage()
+func writeRecords(pgConnection *sql.DB, ksURL string, ksConnection *websocket.Conn) error {
+	_, messageBytes, err := ksConnection.ReadMessage()
+
```
[the remark this answers is not in the document]
Comment (ericatcorsha) [reply collector/collector.go]: done

File: collector/collector.go (hunk 1 above)
Comment (ericatcorsha) [reply collector/collector.go]: done

**Review by DavidCorsha — APPROVED** [review]
